charts/mlp/tests/deployment_test.yaml

```diff
@@ -0,0 +1,104 @@
+suite: Unit tests for mlp Deployment
+templates:
+  - deployment.yaml
+release:
+  name: my-release
+  namespace: my-namespace
+  revision: 1
+  isUpgrade: true
+tests:
+  - it: should set database secret key and name from values if chart specific db is enabled
+    set:
+      global:
+        postgresqlUsername: mlp-global
+        mlp:
+          postgresqlDatabase: mlp-global
+      postgresql:
+        enabled: true
+        postgresqlDatabase: mlp
+        postgresqlUsername: mlp
+      externalPostgresql:
+        enabled: false
+        address: postgresql
+        username: mlp-ext
+        database: mlp-ext
+    asserts:
+      - isKind:
+          of: Deployment
+      - matchRegex:
+          path: metadata.name
+          pattern: my-release-mlp$
+      - equal:
+          path: spec.template.spec.containers[0].env[0].valueFrom.secretKeyRef.name
+          value: my-release-mlp-postgresql
+      - equal:
+          path: spec.template.spec.containers[0].env[0].valueFrom.secretKeyRef.key
+          value: postgresql-password
+  - it: should set database secret key and name from values if chart specific db is disabled, external db is enabled, secret name & key are set
+    set:
+      global:
+        postgresqlUsername: mlp-global
+        mlp:
+          postgresqlDatabase: mlp-global
+      postgresql:
+        enabled: false
+        postgresqlDatabase: mlp
+        postgresqlUsername: mlp
```

Comment:
it is slightly confusing that we have `postgres.enabled` set to `false` and still need to set some values here

Reply:
Sure, I can remove these, Used similar config settings across tests to look consistent/

Reply:
removed